src/api/client/sync/v3.rs

```diff
@@ -382,7 +382,16 @@ async fn process_presence_updates(
 		.ready_fold(PresenceUpdates::new(), |mut updates, (user_id, event)| {
 			match updates.entry(user_id.into()) {
 				| Entry::Vacant(slot) => {
-					slot.insert(event);
+					let mut new_event = event;
+					new_event.content.last_active_ago = match new_event.content.currently_active {
+						| Some(true) => None,
+						| _ => new_event
+							.content
+							.last_active_ago
+							.or(new_event.content.last_active_ago),
```

Comment:
This `or()` is duplicated. I'll fix this altogether with the below review later to avoid unnecessary github actions :-)